docs/development/local-development.md

```diff
@@ -15,16 +15,20 @@ and `exports`. We will transpile them to `commonjs` on build.
 
 #### Prerequisites
 
-For local development some dependencies are required.
+For local development some dependencies are required:
 
 - git
 - nodejs `^10.13.0 || ^12.0.0`
 - yarn `^1.17.0`
 - c++ compiler
 - python `^2.7` with `mock` library
+- java between `8` and `11` (or set env `SKIP_JAVA_TESTS` to `true`)
```

Comment:
Your previous PR showed that Java 13 was fine for Gradle 6. Maybe the easiest thing is to just specify Java 8+?

Reply:
I'm running Java 13 on OSX and just found a failed test due to this. Is it not possible to run the Gradle 5 test with Java 13?

Reply:
I see:

```
const gradleJavaVersionSupport = {
  5: { min: 8, max: 12 },
  6: { min: 8, max: 13 },
};
```

Reply:
exactly.

Reply:
I know this is a documentation PR, but could it be possible to automatically skip those tests if the Java version isn't compatible? I don't want to cause friction to first-time contributors in particular. If we want to make sure we don't end up with a regression in CI, then we could essentially flip the logic so that on CI we set a variable `ENFORCE_JAVA_TESTS` that makes sure we never skip them

Reply:
Shouldn't this also mean java between 8 and 12 is ok?

Reply:
> Shouldn't this also mean java between 8 and 12 is ok?

right. I fixed that.

Reply:
About the tests: I had the same thoughts, however, it is not that easy because jest does not allow to skip a test once it has started running.

I saw the following options:

- Evaluate the Java version before running the tests
   - Advantage: We get correct skip semantics
   - Disadvantages: 
       - Major: we execute synchronous I/O every time we load tests and before the first test can be run.
       - Minor: We cannot use `it.each` but have to use a `for`-loop instead.
 - Simply make the tests succeed if the Java version is wrong
   - Advantage: It works without synchronous I/O on test load
   - Disadvantages: 
       - jest marks the snapshots as obsolete and exists with a non-zero exit code
       - it conveys false information, i.e. the tests are all green even if there might be an error

The proposal with `ENFORCE_JAVA_TESTS` works, of course, but means that people won’t execute the tests by default.

Neither of these options seemed good to me. Because of that, I chose to fail the tests and produce helpful error messages that explain what’s going on. I assumed that most developers will have some version of Java anyway. But that is, of course, only a guess.

Reply:
Your logic is good but I think the biggest problem is that if developers are running Java 13 then they get failed tests. I wouldn't be as concerned if people running *older* versions had to opt out, but it's not like we should expect people to downgrade Java. Even though I helped merge the PR in question, I was still "surprised" the first time I had local test failures. These types of failures can mean surprising friction for contributors because 1 or 2 other things had gone wrong prior, they might just think this project isn't for them.

Reply:
Okay, so which solution do you prefer?

 * Check the Java version before running any tests and skip the test cases accordingly
 * Make the Java tests opt-in via `ENFORCE_JAVA_TESTS`
 * Make the Java tests succeed on a wrong Java version (and live with jest’s reports about obsolete snapshots)

Reply:
I think I prefer number 1, although I'd prefer if we could make sure they're always being run on CI. Earlier you noted a "synchronous" load, but could it be done with await/async instead so that it's not blocking?

Reply:
> but could it be done with await/async instead so that it's not blocking

Nope, describe blocks are always synchronous: https://github.com/facebook/jest/issues/2235

Reply:
Could be combine them within an `it`, or use `beforeAll`, etc? And if we don't, can we quantify what the impact is?

Reply:
I have now changed the behaviour so that test cases are skipped if the Java version does not match or if there is no Java at all.

For CI, I have introduced the `FAIL_IF_NO_JAVA` env to make it more difficult to skip the tests in CI.

Reply:
> Could be combine them within an `it`, or use `beforeAll`, etc

I’m not sure what you have in mind. Note that there does not seem to be any way to skip the test once we have entered the `it` lambda.

> can we quantify what the impact is

Surprisingly, I don’t notice any delay when running the test on my machine with this branch (which now tests for Java synchronously). So it’s probably fine.